File: backend/java/Questions.java
public class Questions {
    private String[][] questions = {
        // Level 4 (Normal)
        {
            "What is the capital of France?",
            "What is 2 + 2?",
            "What is the largest planet in our solar system?"
        },
        // Level 5 (Medium)
        {
            "What is the chemical symbol for gold?",
            "Who wrote 'Romeo and Juliet'?",
            "What is the powerhouse of the cell?"
        },
        // Level 6 (Hard)
        {
            "What is the integral of x^2?",
            "What is the speed of light in vacuum?",
            "Who developed the theory of general relativity?"
        }
    };

    private String[][] answers = {
        // Level 4 (Normal)
        {
            "Paris",
            "4",
            "Jupiter"
        },
        // Level 5 (Medium)
        {
            "Au",
            "William Shakespeare",
            "Mitochondria"
        },
        // Level 6 (Hard)
        {
            "x^3/3 + C",
            "299,792,458 m/s",
            "Albert Einstein"
        }
    };

    public String getQuestion(int level, int index) {
        return questions[level - 4][index];
    }

    public String getAnswer(int level, int index) {
        return answers[level - 4][index];
    }

    public int getNumberOfQuestions(int level) {
        return questions[level - 4].length;
    }
}

File: backend/cpp/questions.cpp
#include <iostream>
#include <vector>
#include <string>

class Question {
public:
    std::string questionText;
    std::vector<std::string> options;
    int correctAnswerIndex;

    Question(std::string text, std::vector<std::string> opts, int correctIndex) 
        : questionText(text), options(opts), correctAnswerIndex(correctIndex) {}
};

class Questions {
private:
    std::vector<Question> level4Questions;
    std::vector<Question> level5Questions;
    std::vector<Question> level6Questions;

public:
    Questions() {
        // Level 4 Questions
        level4Questions.push_back(Question("What is the capital of France?", {"Paris", "London", "Berlin", "Madrid"}, 0));
        level4Questions.push_back(Question("What is 2 + 2?", {"3", "4", "5", "6"}, 1));

        // Level 5 Questions
        level5Questions.push_back(Question("What is the largest planet in our solar system?", {"Earth", "Mars", "Jupiter", "Saturn"}, 2));
        level5Questions.push_back(Question("What is the chemical symbol for water?", {"H2O", "O2", "CO2", "NaCl"}, 0));

        // Level 6 Questions
        level6Questions.push_back(Question("What is the derivative of x^2?", {"2x", "x^2", "x", "2"}, 0));
        level6Questions.push_back(Question("What is the value of pi to two decimal places?", {"3.14", "3.15", "3.16", "3.17"}, 0));
    }

    std::vector<Question> getQuestions(int level) {
        if (level == 4) return level4Questions;
        else if (level == 5) return level5Questions;
        else if (level == 6) return level6Questions;
        else return {};
    }
};